pkg/apis/kubeflow/v2beta1/types.go

```diff
@@ -130,3 +130,116 @@ const (
 	MPIImplementationOpenMPI MPIImplementation = "OpenMPI"
 	MPIImplementationIntel   MPIImplementation = "Intel"
 )
+
+// JobStatus represents the current observed state of the training Job.
+type JobStatus struct {
+	// Conditions is an array of current observed job conditions.
+	// +options
+	Conditions []JobCondition `json:"conditions,omitempty"`
+
+	// ReplicaStatuses is map of ReplicaType and ReplicaStatus,
+	// specifies the status of each replica.
+	// +options
+	ReplicaStatuses map[MPIReplicaType]*ReplicaStatus `json:"replicaStatuses,omitempty"`
```

Comment:
another violation of API conventions T_T
But we can only change it in a new API version.

Reply:
Yes, that's right :(
For v2beta1, we need to keep using this API for compatibility.